server/src/test/TestHelpers.ts

```diff
@@ -0,0 +1,45 @@
+import { CharStreams, CommonTokenStream } from 'antlr4ts';
+import { TextDocument } from 'vscode-languageserver';
+
+import { YmlCompletionItemsProvider } from '../completion/YmlCompletionItemsProvider';
+import { YmlDefinitionProvider } from '../definitions/YmlDefinitionProvider';
+import { YmlLexer, YmlParser } from '../grammar';
+import { YmlFunctionVisitor } from '../visitors/YmlFunctionVisitor';
+
+export const FAKE_FILE_PATH = '/file1.kao';
+export const YML_LANGUAGE_ID = 'yml';
+
+export function createFakeFunctionContainer(functionContent: string = ''): string {
+    return `function myFunction(Object input)
+--> domains Integer
+--> action {
+    ${functionContent}
+};`;
+}
+
+/**
+ * Create and return a TextDocument with `docContent` as content.
+ *
+ * @param docContent the document content
+ *
+ * @return a TextDocument
+ */
+export function createFakeDocument(docContent: string): TextDocument {
+    return TextDocument.create(FAKE_FILE_PATH, YML_LANGUAGE_ID, 1, docContent);
+}
+
+export function parseAndVisitYmlFunction(
```

Comment:
Specifically for a function?
If all `—Visitor` classes have similar args for their constructors, could this tool be parameterized with a type in order to use any visitor you want or something? Dunno.

Reply:
I'll try that.